### DISCUSSION TITLE
Where does Taiga get the different names of Anime from?

### DISCUSSION THREAD

**Discussion by ThePoi** [erengy/taiga#1176] [Questions]
A show can have different names and Taiga does most of the time recognize it. But the new season of MASHLE won't be recognized with only "MASHLE" as the name. I remember seeing that Taiga pulls all Anime names from some site but can not find it. What was it?

**Comment by erengy** [comment]
Taiga gets alternative titles from the active service (AniList, Kitsu or MyAnimeList) rather than an external site. For instance, AniList provides the following ones for MASHLE:

- MASHLE: MAGIC AND MUSCLES
- マッシュル-MASHLE-
- MASHLE ศึกโลกเวทมนตร์คนพลังกล้าม
- MASHLE: MAGIA E MÚSCULOS
- MASHLE: Магия и мускулы

Note that there's also [anime-relations](https://github.com/erengy/anime-relations), which provides redirection rules such as https://github.com/erengy/anime-relations/commit/ff2bc88c0fce14192554ff8fe57a14da6f692eda so that some releases get recognized properly. But it works on IDs and episode numbers rather than names/titles.
